src/goto-programs/remove_calls_nobody.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************\
+
+Module: Remove calls to functions without a body
+
+Author: Daniel Poetzl
+
+\*******************************************************************/
+
+/// \file
+/// Remove calls to functions without a body
+
+#include "remove_calls_nobody.h"
+
+void remove_calls_nobodyt::remove_call_nobody(
+  goto_programt &dest,
+  goto_programt::targett target,
+  const exprt &lhs,
+  const exprt::operandst &arguments)
+{
+  assert(target->is_function_call());
+  assert(!dest.empty());
+
+  goto_programt tmp;
+
+  // evaluate function arguments -- they might have
+  // pointer dereferencing or the like
+  forall_expr(it, arguments)
+  {
+    goto_programt::targett t=tmp.add_instruction();
+    t->make_other(code_expressiont(*it));
```

Comment:
This is probably me not understanding something - but what does making an other expression do? Don't we need some kind of nondet modification if it is a pointer?